src/game/behaviors/blue_fish.inc.c

```diff
@@ -1,76 +1,131 @@
+ * @file blue_fish.inc.c
+ * Implements behaviour and spawning for bhvBlueFish located in the castle aquarium outside of SA.
+ */
 
+/**
+ * Determines fish movement.
+ */
 void bhv_blue_fish_loop(void) {
-    f32 sp24;
+    f32 randomSwitch;
     switch (o->oAction) {
-        case 0:
+        
+        // Initial phase after spawning
+        case BLUE_FISH_SPAWN:
             func_8029ED98(0, 1.0f);
+            
+            // If oTimer equals zero, then assign random values to floats and rotate fish.
             if (o->oTimer == 0) {
-                o->oBlueFishUnk100 = RandomSign() << 11;
-                o->oBlueFishUnkF4 = RandomFloat() * 2;
-                o->oBlueFishUnkF8 = (s32)(RandomFloat() * 30) & 0xFE;
-                sp24 = RandomFloat() * 5;
-                if (sp24 < 2.0f)
+                o->oBlueFishRandomAngle = RandomSign() << 11;
+                o->oBlueFishRandomVel = RandomFloat() * 2;
+                o->oBlueFishRandomTime = (s32)(RandomFloat() * 30) & 0xFE;
+                
+                // Adjusts pitch velocity or sets to zero dependant on outcome of randomSwitch.
+                randomSwitch = RandomFloat() * 5;
+                if (randomSwitch < 2.0f) {
                     o->oAngleVelPitch = random_f32_around_zero(128);
-                else
+                } else {
                     o->oAngleVelPitch = 0;
+                }
             }
-            o->oForwardVel = o->oBlueFishUnkF4 + 3.0f;
-            if (o->oTimer >= o->oBlueFishUnkF8 + 60)
+            
+            // Set forward velocity and progress oAction
+            o->oForwardVel = o->oBlueFishRandomVel + 3.0f;
+            if (o->oTimer >= o->oBlueFishRandomTime + 60) {
                 o->oAction++;
-            if (o->oTimer < (o->oBlueFishUnkF8 + 60) / 2)
+            }
+            
+            // Set pitch velocity
+            if (o->oTimer < (o->oBlueFishRandomTime + 60) / 2) {
                 o->oFaceAnglePitch += o->oAngleVelPitch;
-            else
+            } else {
                 o->oFaceAnglePitch -= o->oAngleVelPitch;
+            }
+            
+            // Calculate new Y velocity
             o->oVelY = -sins(o->oFaceAnglePitch) * o->oForwardVel;
             break;
-        case 1:
+        
+        // Animates and adjusts fish yaw angle, then progresses oAction.
+        case BLUE_FISH_RANDOM_YAW:
             func_8029ED98(0, 2.0f);
-            o->oMoveAngleYaw = (s32)(o->oBlueFishUnk100 + o->oMoveAngleYaw);
-            if (o->oTimer == 15)
+            o->oMoveAngleYaw = (s32)(o->oBlueFishRandomAngle + o->oMoveAngleYaw);
+            if (o->oTimer == 15) {
                 o->oAction++;
+            }
             break;
-        case 2:
+            
+        // Animates and adjusts pitch velocity.
+        case BLUE_FISH_RANDOM_PITCH:
             func_8029ED98(0, 1.0f);
-            if (o->oTimer >= o->oBlueFishUnkF8 + 60)
+            
+            // Progresses oActioon after some time. 
```

Comment:
Typo. Also I'll use this comment to say that "Progresses oAction" is a useless comment, since everyone knows what ++ means. It would be much more helpful to say what action is being progressed to, i.e. `// Switches to BLUE_FISH_ACT_RANDOM_PITCH`